File: src/main/java/util/PyScript.java
package util;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import lombok.experimental.UtilityClass;
import lombok.extern.log4j.Log4j2;

@Log4j2
@UtilityClass
public class PyScript {

  public static void run() throws IOException, InterruptedException {
    ProcessBuilder builder = new ProcessBuilder("python", "src/main/java/util/pypkg/ai.py");
    builder.redirectErrorStream(true);
    Process process = builder.start();

    BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
    String line;
    while ((line = reader.readLine()) != null) {
      log.info(line);
    }
    process.waitFor();
  }
}

File: src/main/java/util/PyScript.kt
package util

import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import java.io.IOException

object PyScript {
  private val log: Logger = LogManager.getLogger(PyScript::class.java)

  fun run() {
    try {
      val builder = ProcessBuilder("python", "src/util.main/java/util/pypkg/ai.py")
      builder.redirectErrorStream(true)
      val process = builder.start()

      process.inputStream.bufferedReader().use {reader ->
        var line: String
        while (reader.readLine().also {line = it} != null) {
          log.info(line)
        }
      }
      process.waitFor()
    } catch (e: IOException) {
      log.error("Failed to execute Python script", e)
    } catch (e: InterruptedException) {
      log.error("Interrupted while waiting for Python script to finish", e)
    }
  }
}
